src/appshell/view/appmenumodel.cpp

```diff
@@ -86,185 +108,179 @@ INotationInteractionPtr AppMenuModel::notationInteraction() const
 {
     MenuItemList editItems {
-        makeAction("undo", canUndo()),
-        makeAction("redo", canRedo()),
+        makeAction("undo"),
+        makeAction("redo"),
         makeSeparator(),
-        makeAction("cut", selectedElementOnScore()),
-        makeAction("copy", selectedElementOnScore()),
-        makeAction("paste", selectedElementOnScore()),
-        makeAction("paste-half", selectedElementOnScore()),
-        makeAction("paste-double", selectedElementOnScore()),
-        makeAction("paste-special", selectedElementOnScore()),
-        makeAction("swap", selectedElementOnScore()),
-        makeAction("delete", selectedElementOnScore()),
+        makeAction("cut"),
+        makeAction("copy"),
+        makeAction("paste"),
+        makeAction("paste-half"),
+        makeAction("paste-double"),
+        makeAction("paste-special"),
+        makeAction("swap"),
+        makeAction("delete"),
         makeSeparator(),
-        makeAction("select-all", scoreOpened()),
-        makeAction("select-similar", selectedElementOnScore()),
-        makeAction("find", scoreOpened()),
+        makeAction("select-all"),
+        makeAction("select-similar"),
+        makeAction("find"),
         makeSeparator(),
         makeAction("preference-dialog") // need implement
     };
 
     return makeMenu(trc("appshell", "&Edit"), editItems);
 }
 
-MenuItem AppMenuModel::viewItem()
+MenuItem AppMenuModel::viewItem() const
 {
     MenuItemList viewItems {
-        makeAction("toggle-palette", isNotationPage(), notationPageState()->isPanelVisible(PanelType::Palette)),
-        makeAction("toggle-instruments", isNotationPage(), notationPageState()->isPanelVisible(PanelType::Instruments)),
-        makeAction("masterpalette", isNotationPage(), paletteController()->isMasterPaletteOpened().val),
-        makeAction("inspector", isNotationPage(), notationPageState()->isPanelVisible(PanelType::Inspector)),
-        makeAction("toggle-playpanel", isNotationPage()), // need implement
-        makeAction("toggle-navigator", isNotationPage(), notationPageState()->isPanelVisible(PanelType::NotationNavigator)),
-        makeAction("toggle-timeline", isNotationPage()), // need implement
-        makeAction("toggle-mixer", isNotationPage()), // need implement
-        makeAction("synth-control", isNotationPage()), // need implement
-        makeAction("toggle-selection-window", isNotationPage()), // need implement
-        makeAction("toggle-piano", isNotationPage()), // need implement
-        makeAction("toggle-scorecmp-tool", isNotationPage()), // need implement
+        makeAction("toggle-palette", isPanelVisible(PanelType::Palette)),
+        makeAction("toggle-instruments", isPanelVisible(PanelType::Instruments)),
+        makeAction("masterpalette", paletteController()->isMasterPaletteOpened().val),
+        makeAction("inspector", isPanelVisible(PanelType::Inspector)),
+        makeAction("toggle-navigator", isPanelVisible(PanelType::NotationNavigator)),
```

Comment:
Seems that you can add a method to IMenuController:

actionActivated(actionCode)

The goal of this method to tell us that user activated this action (you can pick a more proper name for it). So, makeAction will look like this:

makeAction(actionCode) {
...
item.enabled = resolveController(actionCode)->actionEnabled(actionCode)
item.checked = resolveController(actionCode)->actionActivated(actionCode)
}